src/main/java/uk/gov/justice/digital/hmpps/cmd/api/domain/FullDayActivityType.kt

```diff
@@ -20,10 +20,16 @@ enum class FullDayActivityType(val description: String) {
   ;
 
   companion object {
-    fun from(value: String): FullDayActivityType {
-      return Arrays.stream(values())
-        .filter { type -> value.contains(type.description, true) }
-        .findFirst().orElse(SHIFT)
-    }
+    fun from(value: String?): FullDayActivityType =
+      if (value == null) {
+        log.warn("FullDayActivityType: Activity is missing - possibly using deleted Day Model")
+        SHIFT
+      } else {
+        entries
+          .find { type -> value.contains(type.description, true) }
+          ?: SHIFT
+      }
+
```

Comment:
BTW if the logging was important you could have written

`value?..find { type -> value.contains(type.description, true) ?: SHIFT`